### PR TITLE
grass.script: Add available_cpus and resolve_nprocs helpers

### PR DESCRIPTION
**Pull request by petrasovaa**
We are using `G_set_omp_num_threads` that resolves nprocs <= 0, but in Python, we don't have equivalent, potentially resulting in bugs. This PR adds the function for resolving the nprocs and also function for getting available cpus that works better in containers.

  - **`available_cpus()`** — number of CPUs this process may actually use. Prefers affinity-aware sources
  (`os.process_cpu_count` on 3.13+, `os.sched_getaffinity` on Linux) over `os.cpu_count`, which reports the host total
   and overcounts in containers and cgroup-limited jobs.
  - **`resolve_nprocs(nprocs)`** — mirrors `G_set_omp_num_threads()` in `lib/gis/omp_threads.c`:
    - positive → used as-is
    - `0` → all available cores
    - negative → `available + nprocs`, clamped to at least 1

These functions are applied in the code base, resulting in some bug fixes:

  **`t.rast.univar` / `t.rast3d.univar`**: `nprocs=0` (documented as "use all available cores") was silently clamped  to 1, so the tool ran serially. The tool entry points now apply `gs.resolve_nprocs()` so the documented  `G_OPT_M_NPROCS` semantics take effect for `0` and negative values.

AI-assisted (Claude Code) — used for migration planning, edge case analysis, and drafting docstrings and tests.

### CODE REVIEW

**Review by cwhite911 — APPROVED** [review]

File: python/grass/pygrass/modules/interface/module.py [hunk 1]
```diff
@@ -231,12 +231,14 @@ class ParallelModuleQueue:
     def __init__(self, nprocs=1):
         """Constructor
 
-        :param nprocs: The maximum number of Module processes that
-                       can be run in parallel, default is 1, if None
-                       then use all the available CPUs.
+        :param nprocs: The maximum number of Module processes that can be
+                       run in parallel. A positive value is used as-is,
+                       ``0`` or ``None`` means use all available CPUs, and
+                       a negative value reserves that many CPUs (clamped
```
Comment (wenzeslaus) [inline python/grass/pygrass/modules/interface/module.py:237]: This is.not clear. I don't see how it explains the behavior.

File: python/grass/script/utils.py [hunk 2]
```diff
@@ -93,6 +93,40 @@ def separator(sep: str) -> str:
...
+        return len(os.sched_getaffinity(0))
+    return os.cpu_count() or 1
+
+
+def resolve_nprocs(nprocs: int | str) -> int:
+    """Resolve G_OPT_M_NPROCS into a worker count.
+
+    Mirrors the semantics of ``G_set_omp_num_threads()`` in
+    ``lib/gis/omp_threads.c``: 0 means use all available cores, a positive
+    number is used as-is, a negative number means cpu_count + nprocs
```
Comment (wenzeslaus) [inline python/grass/script/utils.py:116]: This is correct, but maybe also explaining the actual usage would be helpful  (keep that many available, use the rest).

**Review by wenzeslaus — COMMENTED** [review]
I had these minor comments pending. Please, consider them for your next related PR.
